CMakeLists.txt

```diff
@@ -262,6 +262,7 @@ if(BUILD_TESTING)
 endif()
 
 configure_file(config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
+configure_file(metainfo.xml.in ${CMAKE_CURRENT_BINARY_DIR}/${APPLICATION_REV_DOMAIN}.metainfo.xml)
```

Comment:
this should go next to the desktop file since it should only be used on linux
https://github.com/nextcloud/desktop/blob/20357ca7fa5d740bd6b76f18f76e7098f30863d5/src/gui/CMakeLists.txt#L437-L439
and it should also get installed like the desktop file, so some like this:
```cmake
configure_file(${CMAKE_SOURCE_DIR}/metainfo.xml.in 
               ${CMAKE_CURRENT_BINARY_DIR}/${LINUX_APPLICATION_ID}.metainfo.xml) 
install(FILES  ${CMAKE_CURRENT_BINARY_DIR}/${LINUX_APPLICATION_ID}.metainfo.xml DESTINATION ${DATADIR}/metainfo )
```